src/Dropdown/Dropdown.js

```diff
@@ -85,26 +95,25 @@ export default function Dropdown({
   useKeyPress(ARROW_KEYS, event => {
     if (isOpen && menuRef.current) {
       event.preventDefault();
+      event.stopPropagation();
       const focusArgs = [menuRef.current, document.activeElement];
       const nextFocusable =
         event.key === 'ArrowUp' ? findPreviousFocusableElement(...focusArgs) : findNextFocusableElement(...focusArgs);
 
       if (nextFocusable) {
+        setBlurable(false);
         nextFocusable.focus();
+        setBlurable(true);
       }
     }
   });
 
-  const handleMenuBlur = () => {
-    if (autoclose) {
-      // Use timeout to delay examination of activeElement until after blur/focus
-      // events have been processed.
-      setTimeout(() => {
```

Comment:
This setTimeout was causing a bug with blur when inputs were clicked. Managing whether blur is being triggered via the state now allows inputs clicks to trigger blur.